src/main/java/uk/gov/register/service/SpreadSheetTransformerService.java

```diff
@@ -0,0 +1,97 @@
+        final List<String> fieldNames = new ArrayList<>();
+        final List<Map<String, String>> elements = new ArrayList<>();
+
+        fieldNames.add("index-entry-number");
+        fieldNames.add("entry-number");
+        fieldNames.add("entry-timestamp");
+        fieldNames.add("key");
+
+        recordsView.getFields().forEach(field -> fieldNames.add(field.getFieldName()));
+
+        recordsView.getRecords()
+                .forEach((key, value) -> {
+                    final Map<String, String> element = new HashMap<>();
+
+                    element.put("index-entry-number", key.getIndexEntryNumber().toString());
+                    element.put("entry-number", key.getEntryNumber().toString());
+                    element.put("entry-timestamp", key.getTimestampAsISOFormat());
+                    element.put("key", key.getKey());
+
+                    value.forEach(v -> v.getContent()
+                            .forEach((contentKey, contentValue) -> element.put(contentKey, contentValue.getValue())));
+
+                    elements.add(element);
+                });
+
+        toSpreadSheet(outputStream, registerName, fieldNames, elements);
+    }
+
+    public void entriesToSpreadSheet(final OutputStream outputStream, final String registerName, final Collection<Entry> entries) {
+        final List<String> fieldNames = new ArrayList<>();
+        final List<Map<String, String>> elements = new ArrayList<>();
+
+        fieldNames.add("index-entry-number");
+        fieldNames.add("entry-number");
+        fieldNames.add("entry-timestamp");
+        fieldNames.add("key");
+        fieldNames.add("item-hash");
+
+        entries.forEach(entry -> {
+            final Map<String, String> element = new HashMap<>();
+
+            element.put("index-entry-number", entry.getIndexEntryNumber().toString());
+            element.put("entry-number", entry.getEntryNumber().toString());
+            element.put("entry-timestamp", entry.getTimestampAsISOFormat());
+            element.put("key", entry.getKey());
+            element.put("item-hash", entry.getItemHashes().get(0).toString());
+
+            elements.add(element);
+        });
+
+        toSpreadSheet(outputStream, registerName, fieldNames, elements);
+    }
+
+    public void itemToSpreadSheet(final OutputStream outputStream, final String registerName, final ItemView itemView) {
+        final List<String> fieldNames = new ArrayList<>();
+        final List<Map<String, String>> elements = new ArrayList<>();
+        final Map<String, String> element = new HashMap<>();
+
+        itemView.getFields().forEach(field -> fieldNames.add(field.getFieldName()));
+
+        itemView.getContent().forEach((fieldName, fieldValue) -> element.put(fieldName, fieldValue.getValue()));
+
+        elements.add(element);
+
+        toSpreadSheet(outputStream, registerName, fieldNames, elements);
+    }
+
+    public String getFilename(final String identifier) {
```

Comment:
If we made the endpoint `/*. xlsx` would we need to do this? If we follow our other downloads at the moment, none of them have any information regarding register / timestamp.

Reply:
We want to added, right now, if you download multiple files, yo don't have any clue about what they are if you don't open them.